### ISSUE TITLE
stuck on obtaining IP address

### ISSUE THREAD

**Issue by Stanwarr** [wifiphisher/wifiphisher#1093]
**Version**: c27c2ab1ef9c5c3aad6f9643acb9170cf2c45a73

**Description**: Everything works fine except the fact that I can't connect to the fake AP (stuck on obtaining IP address). Seems like it fails to obtain an IP through DHCP. Note that my adapter supports AP mode.
**Script output**: Nothing interesting

**Comment by sophron** [comment]
Can you verify that `dnsmasq` is running? What is the output of `ps -aux | grep dnsmasq` and `cat /tmp/dnsmasq.conf` while Wifiphisher is running?

**Comment by Stanwarr** [comment]
> Can you verify that `dnsmasq` is running? What is the output of `ps -aux | grep dnsmasq` and `cat /tmp/dnsmasq.conf` while Wifiphisher is running?


Output: `root      5734  0.0  0.0  17484   892 pts/1    S+   00:28   0:00 grep --color=auto dnsmasq
`
`root@Stanwar:/home/stan# cat /tmp/dnsmasq.conf
no-resolv
interface=wlp3s0
dhcp-range=192.168.1.2,192.168.1.100,12h
address=/google.com/172.217.5.78
address=/clients3.google.com/172.217.11.174
address=/#/192.168.1.1 root@Stanwar:/home/stan# 
`
I edited dnsmasq through constants in an attempt to fix it, but that didn't help.

**Comment by sophron** [comment]
It appears that dnsmasq is not running in your system. What will happen if you type `dnsmasq /etc/dnsmasq.conf` when you run Wifiphisher on a different window?

**Comment by Stanwarr** [comment]
Running that command before running wifiphisher is successful which makes me able to connect to the fake AP later on when I launch wifiphisher. Though wifiphisher shows some errors relating to the fake web page ports being used. And when I open any web page, whether http or https, it simply shows connection error. @sophron

**Comment by sophron** [comment]
If these are SSL connection errors, they are most likely normal.

I'm still curious to know why `dnsmasq` fails to be executed from Wifiphisher but works when you run it on your own. Do you get any additional errors that could help us troubleshoot this?

**Comment by Stanwarr** [comment]
> If these are SSL connection errors, they are most likely normal.
> 
> I'm still curious to know why `dnsmasq` fails to be executed from Wifiphisher but works when you run it on your own. Do you get any additional errors that could help us troubleshoot this?

Not SSL connection errors. More like I can't access any web page. Yes, there's something worth mentioning that is there's issue with the port 53. I get this when I install dnsmasq:
`Setting up dnsmasq (2.79-1) ...
Job for dnsmasq.service failed because the control process exited with error code.
See "systemctl status dnsmasq.service" and "journalctl -xe" for details.
invoke-rc.d: initscript dnsmasq, action "start" failed.
● dnsmasq.service - dnsmasq - A lightweight DHCP and caching DNS server
   Loaded: loaded (/lib/systemd/system/dnsmasq.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2019-03-20 16:10:29 CET; 5ms ago
  Process: 4943 ExecStart=/etc/init.d/dnsmasq systemd-exec (code=exited, status=2)
  Process: 4929 ExecStartPre=/usr/sbin/dnsmasq --test (code=exited, status=0/SUCCESS)

مارس 20 16:09:59 Stanwar systemd[1]: Starting dnsmasq - A lightweight DHCP and caching DNS server...
مارس 20 16:10:22 Stanwar dnsmasq[4929]: dnsmasq: syntax check OK.
مارس 20 16:10:29 Stanwar dnsmasq[4943]: dnsmasq: failed to create listening socket for port 53: Address already in use
مارس 20 16:10:29 Stanwar systemd[1]: dnsmasq.service: Control process exited, code=exited status=2
مارس 20 16:10:29 Stanwar dnsmasq[4943]: failed to create listening socket for port 53: Address already in use
مارس 20 16:10:29 Stanwar systemd[1]: dnsmasq.service: Failed with result 'exit-code'.
مارس 20 16:10:29 Stanwar dnsmasq[4943]: FAILED to start up
مارس 20 16:10:29 Stanwar systemd[1]: Failed to start dnsmasq - A lightweight DHCP and caching DNS server.
`

**Comment by sophron** [comment]
Apparently, there is something already listening on port 53. Try `lsof -Pn +M | grep ':53 (LISTEN)'` to find that process.

**Comment by Stanwarr** [comment]
> Apparently, there is something already listening on port 53. Try `lsof -Pn +M | grep ':53 (LISTEN)'` to find that process.

`tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      102        143934     10164/systemd-resol `
Seems like there's a conflict between systemd-resol and dnsmasq

**Comment by cyablo** [comment]
Same problem for me (Kali Linux) but nothing blocking Port 53. dnsmasq starts with wifiphisher but does not seem to work.

**Comment by sophron** [comment]
@cyablo That's interesting. Is it a specific device that it fails to connect?

**Comment by sophron** [comment]
@Stanwarr We should ensure that port 53 is available. I'm adding the "Enhancement" tag.

**Comment by Stanwarr** [comment]
> @cyablo That's interesting. Is it a specific device that it fails to connect?

No. I tried it on different computer/different adapters (I installed ubuntu on a USB drive so it's easy for me to do so). 
So here's the issue: There's a conflict between systemd-resolved and dnsmasq. We should disabled systemd-resolved and use dnsmasq system-wide. And it seems like even when I disabled systemd-resolved, I have to manually kill dnsmasq then start it again using `dnsmasq -C \tmp\dnsmasq.conf` before opening wifiphisher in order for it to work.

**Comment by cyablo** [comment]
> @cyablo That's interesting. Is it a specific device that it fails to connect?

No, tried two Android phones (HTC and OnePlus). Booth show the same behavior. Also tried multiple WiFi Adapters (Ralink & Atheros). Other tools like Airgeddon and Fluxion work just fine. So I it does not seem to be the Adapter or dnsmasq itself.

**Comment by haroonalhadisk** [comment]
@cyablo try editing the /etc/dnsmasq.conf and set them to default value
here is alink to the default values : https://oss.segetech.com/intra/srv/dnsmasq.conf
then use "dnsmasq /etc/dnsmasq.conf" in another terminal.
 had the same issue fixed it once i did these things. try if it works for you.

**Comment by dipakvz** [comment]
actual problem is with port 53. disable & stop "systemd-resolved.service" with following command.
$ sudo systemctl disable systemd-resolved.service
$ sudo systemctl stop systemd-resolved
I am using Pop_os 20.04 and it worked for me. I tried on Linux Mint and Elementary Os, in that also worked for me.
